rafthttp/stream.go

```diff
@@ -392,45 +392,52 @@ func (cr *streamReader) dial(t streamType) (io.ReadCloser, error) {
 	lv := semver.Must(semver.NewVersion(version.Version))
 	if compareMajorMinorVersion(rv, lv) == -1 && !checkStreamSupport(rv, t) {
 		resp.Body.Close()
+		cr.picker.unreachable(u)
 		return nil, errUnsupportedStreamType
 	}
 
-	switch resp.StatusCode {
-	case http.StatusGone:
+	err = cr.checkResponse(resp)
+	if err != nil {
 		resp.Body.Close()
-		err := fmt.Errorf("the member has been permanently removed from the cluster")
-		select {
-		case cr.errorc <- err:
-		default:
+		cr.picker.unreachable(u)
+		// errMemberRemoved is a critical error since a removed member should
+		// always be stopped. So we use reportCriticalError to report it to errorc.
+		if err == errMemberRemoved {
+			reportCriticalError(err, cr.errorc)
 		}
 		return nil, err
+	}
+
+	return resp.Body, nil
+}
+
+func (cr *streamReader) checkResponse(resp *http.Response) error {
```

Comment:
The function may also read the body of the response, so `HTTPStatus` is not accurate. `checkHTTPResponse` could be a better naming. I will add comment for this function to describe it.

> Also can we add a few comments? 

sure